lumicks/pylake/detail/confocal.py

```diff
@@ -8,6 +8,25 @@
 from .image import reconstruct_image_sum, reconstruct_image, save_tiff, ImageMetadata
 
 
+def _default_image_factory(self, channel):
+    channel_data = getattr(self, f"{channel}_photon_count").data
+    raw_image = reconstruct_image_sum(channel_data, self.infowave.data, self._shape)
+    return self._to_spatial(raw_image)
+
+
+def _default_timestamp_factory(self, channel):
+    assert channel == "timestamps"
```

Comment:
On second thought, I think that assertion should be in the attribute, since it is actually there to protect the caching mechanism. The factory doesn't even need this argument (I'll kill it).

The caching itself and the assertion weren't added in this PR.